src/redisearch_rs/rlookup/src/lookup.rs

```diff
@@ -265,10 +140,28 @@ impl<T> Default for CBOption<T> {
 /// This is used for data generated on the fly, or for data not stored within
 /// the sorting vector.
 /// ```
+///
+/// cbindgen:no-export
 #[pin_project(!Unpin)]
 #[derive(Debug)]
 #[repr(C)]
 pub struct RLookupKey<'a> {
+    // THIS MUST BE THE FIRST FIELD
+    header: RLookupKeyHeader<'a>,
+
+    /// The actual "owning" strings, we need to hold onto these
+    /// so the pointers in the above header stay valid. Note that you
+    /// MUST NEVER MOVE THESE BEFORE THE name AND path FIELDS UNLESS
+    /// YOU WANT TO POTENTIALLY RISK UB
+    #[pin]
+    _name: Cow<'a, CStr>,
+    #[pin]
+    _path: Option<Cow<'a, CStr>>,
```

Comment:
The `name` and `path` wording still does not make sense here

Reply:
good point yeah